docs/reference/settings.rst

```diff
@@ -246,6 +247,261 @@ subclass.
+
+This setting, if not empty, takes precedence over the
+:setting:`ZYTE_API_SESSION_PARAMS` setting and the
+:reqmeta:`zyte_api_session_params` request metadata key, but it can be
+overridden by the :reqmeta:`zyte_api_session_location` request metadata key.
+
+To disable the :setting:`ZYTE_API_SESSION_LOCATION` setting on a specific
+request, e.g. to use the :setting:`ZYTE_API_SESSION_PARAMS` setting or the
+:reqmeta:`zyte_api_session_params` request metadata key instead, set
+the :reqmeta:`zyte_api_session_location` request metadata key to ``{}``.
+
+
+.. setting:: ZYTE_API_SESSION_MAX_BAD_INITS
+
+ZYTE_API_SESSION_MAX_BAD_INITS
+==============================
+
+Default: ``8``
+
+The maximum number of :ref:`scrapy-zyte-api sessions <session>` per pool that
+are allowed to fail their session check right after creation in a row. If the
+maximum is reached, the spider closes with ``bad_session_inits`` as the close
+reason.
+
+To override this value for specific pools, use
+:setting:`ZYTE_API_SESSION_MAX_BAD_INITS_PER_POOL`.
+
+
+.. setting:: ZYTE_API_SESSION_MAX_BAD_INITS_PER_POOL
+
+ZYTE_API_SESSION_MAX_BAD_INITS_PER_POOL
+=======================================
+
+Default: ``{}``
+
+:class:`dict` where keys are :ref:`pool <session-pools>` IDs and values are
+overrides of :setting:`ZYTE_API_SESSION_POOL_SIZE` for those pools.
+
+
+.. setting:: ZYTE_API_SESSION_MAX_ERRORS
+
+ZYTE_API_SESSION_MAX_ERRORS
+===========================
+
+Default: ``1``
+
+Maximum number of :ref:`unsuccessful responses
+<zyte-api-unsuccessful-responses>` allowed for any given session before
+discarding the session.
+
+You might want to increase this number if you find that a session may continue
+to work even after an unsuccessful response. See :ref:`optimize-sessions`.
+
+.. note:: This setting does not affect session checks
+    (:setting:`ZYTE_API_SESSION_CHECKER`). A session is always discarded the
+    first time it fails its session check.
+
+
+.. setting:: ZYTE_API_SESSION_PARAMS
+
+ZYTE_API_SESSION_PARAMS
+=======================
+
+Default: ``{"browserHtml": True}``
+
+Parameters to use for session initialization.
+
+It works similarly to :http:`request:sessionContextParams` from
+:ref:`server-managed sessions <zyte-api-session-contexts>`, but it supports
+arbitrary Zyte API parameters instead of a specific subset.
+
+If it does not define a ``"url"``, the URL of the request for which the session
+is being initialized will be used.
+
+This setting can be overridden by the :setting:`ZYTE_API_SESSION_LOCATION`
+setting, the :reqmeta:`zyte_api_session_location` request metadata key, or the
+:reqmeta:`zyte_api_session_params` request metadata key.
+
+Example:
+
+.. code-block:: python
+    :caption: settings.py
+
+    ZYTE_API_SESSION_PARAMS = {
+        "browserHtml": True,
+        "actions": [
+            {
+                "action": "setLocation",
+                "address": {"postalCode": "10001"},
```

Comment:
We need a way to inject the location values here without having to hardcode them. For example, "$ZYTE_API_SESSION_LOCATION" is replaced with the content of ZYTE_API_SESSION_LOCATION value, or similar?